src/widget/systemtrayicon.cpp

```diff
@@ -107,16 +111,19 @@ SystemTrayIcon::SystemTrayIcon()
         {
             delete reinterpret_cast<QImage*>(image);
         };
-        QImage image(":/img/icon.png");
-        if (image.format() != QImage::Format_RGBA8888_Premultiplied)
-        image = image.convertToFormat(QImage::Format_RGBA8888_Premultiplied);
-        GdkPixbuf* pixbuf = gdk_pixbuf_new_from_data(image.bits(), GDK_COLORSPACE_RGB, image.hasAlphaChannel(),
-                                 8, image.width(), image.height(),
-                                 image.bytesPerLine(), callbackFreeImage, &image);
+        QImage* image = new QImage(":/img/icon.png");
+        if (image->format() != QImage::Format_RGBA8888_Premultiplied)
+            *image = image->convertToFormat(QImage::Format_RGBA8888_Premultiplied);
+        GdkPixbuf* pixbuf = gdk_pixbuf_new_from_data(image->bits(), GDK_COLORSPACE_RGB, image->hasAlphaChannel(),
```

Comment:
Hm… this cannot be correct in this case. We use QImage here to load and initialize image data, but then transport the data to a gtk object. To me this looks like a pretty dirty hack -> and it crashes, because data isn't copied, but implicitly shared here (see docs for QImage::bits()). This requires a `memcpy(QImage::bits())`. Otherwise image data become invalid for `GdkPixbuf` when QImage is deleted.